### PR TITLE
Fixed the bug that causes the restaurants with same name to not render

### PR DESCRIPTION
**Pull request by chc5**
Updates:
 - Fixed a bug that causes restaurants with same name to not render properly. (e.g. When clicking on one of the McDonald's in the area and then clicking on another McDonald's in the same area will not render the latest restaurant that was clicked)
 - Added a default case when there's no ratings that can be found on a certain unknown restaurant.

Hours Spent: 1

@zhuoyingcai Could you confirm that the Dunkin Donuts' bug is resolved?

### CODE REVIEW

File: client/src/BusinessCard/BusinessCard.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ class BusinessCard extends Component {
     let data = this.retrieveData();
     let urlParams = Object.entries(data).map(e => e.join('=')).join('&');
     this.setState({ loading: true });
+    this.setState({ mainBusiness: null });
```
Comment (sengfung27) [inline client/src/BusinessCard/BusinessCard.js]: the setState should contain these two state instead of calling the setState twice

**Review by sengfung27 — CHANGES_REQUESTED** [review]
Fix that setState and overall is fine,
I will let @zhuoyingcai to see if it's resolved.

File: client/src/BusinessCard/BusinessCard.js (hunk 1 above)
Comment (chc5) [reply client/src/BusinessCard/BusinessCard.js]: Resolved

Comment (sengfung27) [reply client/src/BusinessCard/BusinessCard.js]: Thank you

**Review by sengfung27 — APPROVED** [review]

**Review by zhuoyingcai — APPROVED** [review]
Great job! Reviews would update for restaurants that have the same name.

**Pull request merged**